src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt

```diff
@@ -21,24 +19,31 @@ class AuthenticationFilter: ContainerRequestFilter {
     }
 
     override fun filter(requestContext: ContainerRequestContext) {
-
         if(requestContext.uriInfo.path == LOGIN_PATH) return
 
         val cookies = requestContext.cookies
         val jwtCookie = cookies[COOKIE_NAME]?.value
 
-        verify(jwtCookie)
+        verify(jwtCookie, requestContext)
     }
 
-    fun verify(cookie: String?) {
+    fun verify(cookie: String?, requestContext: ContainerRequestContext) {
+
         if (cookie.isNullOrEmpty())
             throw WebApplicationException(
                 Response.status(Response.Status.UNAUTHORIZED).entity("Invalid Cookie").build()
             )
 
         cookie.let {
             try {
-                Jwts.parser().setSigningKey(SECRET_KEY)
+                val claims = Jwts.parser()
+                                    .setSigningKey(SECRET_KEY)
+                                    .parseClaimsJws(cookie)
+                                    .body
+                requestContext.setProperty("employeeId", claims["id"])
+                requestContext.setProperty("companyId", claims["companyId"])
+                requestContext.setProperty("companyId", claims["companyId"])
```

Comment:
you have companyId twice here, did you mean to put something else?

Reply:
Yes, my mistake. Fixed it